jina/clients/python/request.py

```diff
@@ -87,6 +87,16 @@ def _generate(data: Union[Iterator['jina_pb2.Document'], Iterator[bytes], Iterat
                 queryset = [queryset]
             req.queryset.extend(queryset)
 
+        if top_k and mode == ClientMode.SEARCH:
+            if top_k <= 0:
+                raise ValueError(f'"top_k: {top_k}" is not a valid number')
```

Comment:
I am not sure about the context of this exception, but I read it as: The client can define a value, which will make jina throw an exception. Is this exception catches on a per-request basis? If so, everything is fine. If not, a client can crash jina with a bad request. That would be not good.

Reply:
This works on the client, so jina would, so not necessarily. This exception is taken from an old version where this functionality was present.